src/Servers/Kestrel/Core/src/Internal/Http3/Http3ControlStream.cs

```diff
@@ -249,6 +216,20 @@ private async Task HandleControlStream()
             }
         }
 
+        private async ValueTask HandleEncodingDecodingTask()
+        {
+            // Noop encoding and decoding task. Settings make it so we don't need to read content of encoder and decoder.
+            // An endpoint MUST allow its peer to create an encoder stream and a
+            // decoder stream even if the connection's settings prevent their use.
+
+            while (_isClosed == 0)
```

Comment:
In regard to the comment before: here `_isClosed` is declared as `volatile`.
Also the state is updated by `Interlocked`, so I'd remove the `volatile`-modifier and use explicit `Volatile.Reads` to make it more obvious that it's a volatile read. But this is a matter of taste.